tf-modules/data-persistence/dynamo.tf

```diff
@@ -11,7 +11,6 @@ locals {
     providers_table        = "${var.prefix}-ProvidersTable"
     rules_table            = "${var.prefix}-RulesTable"
     semaphores_table       = "${var.prefix}-SemaphoresTable"
-    users_table            = "${var.prefix}-UsersTable"
```

Comment:
We definitely need to add a `CHANGELOG` entry for removing this table. Possibly a `DATA_CHANGELOG` entry as well?

The README for this module should also be updated. And we might want to do a quick search docs to see if this table is referenced anywhere.

Reply:
I'll defer to @nemreid or @laurenfrederick on that one. My inclination is that, since this should be transparent to the users (and integrators), there wouldn't need to be a note anywhere saying that the table had been removed.

Reply:
I think a `DATA_CHANGELOG` note is in order specifically because this does constitute a breaking change to the implementation of the data model. It's a bit of a special case since we depend on sourcing this information from configs and not previous states of the data, so it does not require a migration, but the `DATA_CHANGELOG` in particular is for *us* to track changes to the data model, not so much for users.

I don't think we need to get into the specific table and such necessarily but users might be served with a note along the lines of 

```
 - API backend solution for storing authorized users has migrated from DynamoDB to S3.
```

as the reality is that users *are* aware of our tables, regardless of whether we would prefer them not to be, and the sudden disappearance of a table might be cause for concern.

Reply:
Added in df58980f2

Reply:
We also need to update the README for the data-persistence module to reflect the removal of this table